fdbserver/MoveKeys.actor.cpp

```diff
@@ -236,6 +236,45 @@ ACTOR Future<vector<UID>> addReadWriteDestinations(KeyRangeRef shard,
 	return result;
 }
 
+// Returns storage servers selected from 'candidates', who is serving a read-write copy of 'range'.
+ACTOR Future<std::vector<UID>> pickReadWriteServers(Transaction* tr, std::vector<UID> candidates, KeyRangeRef range) {
+	vector<Future<Optional<Value>>> serverListEntries;
+
+	for (const UID id : candidates) {
+		serverListEntries.push_back(tr->get(serverListKeyFor(id)));
+	}
+
+	vector<Optional<Value>> serverListValues = wait(getAll(serverListEntries));
+
+	std::vector<StorageServerInterface> ssis;
+	for (auto& v : serverListValues) {
+		ssis.push_back(decodeServerListValue(v.get()));
+	}
+
+	state std::vector<Future<Optional<UID>>> checks;
+	checks.reserve(ssis.size());
+	for (auto& ssi : ssis) {
+		checks.push_back(checkReadWrite(
+		    ssi.getShardState.getReplyUnlessFailedFor(GetShardStateRequest(range, GetShardStateRequest::NO_WAIT),
+		                                              SERVER_KNOBS->SERVER_READY_QUORUM_INTERVAL,
+		                                              0,
+		                                              TaskPriority::MoveKeys),
+		    ssi.id(),
+		    0));
```

Comment:
Is it correct to use 0 here? I'm not very familiar with the code. [Here](https://github.com/halfprice/foundationdb/blob/51aac91456053257143f48ecbdd8ab7da00312fe/fdbserver/MoveKeys.actor.cpp#L207-L217) indicates that the dst needs to wait until reach a certain version. So is the same requirement not needed here?

Reply:
Since all the source servers are lost, I assumed `any data will be better than no data`, but I am open to other options.